### ISSUE TITLE
Hierarchical runs too slow

### ISSUE THREAD

**Issue by cheesebear** [wannesm/dtaidistance#140]
Hierarchical seems to run np.min too many times. I suggest apply stack to dists first, to avoid running min repeatedly.

**Comment by wannesm** [comment]
You are correct that, this method can be further optimized and the usage of `min` is not ideal. Do you have a worked out solution? I assume you imply argsort instead of sort?

For fast clustering implementations I recommend to use the Scipy and Pyclustering implementations. They are already wrapped in `clustering.hierarchical.LinkageTree` and `clustering.medoids.Medoids`, respectively. And by looking at the code in those classes, one can use almost all other clustering techniques in the other toolbox.
